### ISSUE TITLE
Rename operator kind

### ISSUE THREAD

**Issue by fabiand** [kubevirt/kubevirt#2002]
/kind enhancement

**What happened**:
Operator kind is `kubevirt.kubevirt.io`

I'd ask to rename it to `operator.kubevirt.io` or `application.kubevirt.io`.

**Anything else we need to know?**:
`kubevirt.kubevirt.io` seems to be to unspecific.

**Comment by Siddhantrao** [comment]
I would like to work on this , Is there anything you would like me to keep in mind before I start?

**Comment by fabiand** [comment]
@slintes is the one who can give some guidance here.

I still have the impression that we want to improve the naming a bit, maybe looking at how other projects named their CRs, operators, and namespaces.

**Comment by slintes** [comment]
well, tbh, I'm unsure what to do exactly about the name. Let me try to explain.
This is the relevant part of our current CRD:
```
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: kubevirts.kubevirt.io
spec:
  group: kubevirt.io
  names:
    singular: kubevirt
    plural: kubevirts
    kind: KubeVirt
    shortNames:
    - kv
    - kvs
  version: v1alpha3
```
There are some conventions:
- `metadata.name` must match `<spec.names.plural>.<spec.group>`
- `spec.names.kind` normally is CamelCased `spec.name.singular`

So, if you want to change the name to e.g. `application.kubevirt.io`, the singular name would be `application`, and kind would be `Application`.
In manifests you use the kind. No a big problem, because they also contain the group, so it's clear what application you're talking about.
But with `kubectl` you usually only use the singular or plural names, or even the short names (what would they be for `application`?). So you would use e.g. `kubectl get application`. That is something I dislike. You don't see that it's about KubeVirt anymore, and if more operators start using this name it would result in confusion, and I don't want to start this. (You can also use the complete name `application.kubevirt.io` with kubectl, but I think nobody wants to do that).
So long story short: I still think KubeVirt makes most sense as kind, with singular and plural names as they are now, and last but not least with the imho nice short names.
Maybe we can just extend the group to e.g. `application.kubevirt.io`, so the full name is `kubevirt.application.kubevirt.io`, but everything else stays the same?

**Comment by fabiand** [comment]
Hm.
Thanks for the explanation @slintes 

TBH I don#t see how we can make it better, and I also don't see that other projects have neat ideas.
Thus I think we can close this

Sorry @Siddhantrao
